### ISSUE TITLE
Explanation of output files

### ISSUE THREAD

**Issue by samuelhaysom** [martinpacesa/BindCraft#203]
Thanks for all your work on this package! Please point me towards this if it already exists but would you be able to add more explanation of the generated outputs to the README? I'd like to compare failure/success rates of different runs but am confused how to find these. I assume most of what I need is in failure_csv but don't understand what the numbers in here mean. 

I assumed every column header represents a stage/failure reason within the pipeline (e.g. Trajectory_logits_pLDDT meaning designs that failed to meet the pLDDT threshold at the end of the logits stage). But some of the columns have numbers higher than the total number of trajectories I've generated (as indicated by the number of rows in the trajectory_stats file). E.g. i_pTM = 3336 but I only generated 176 trajectories.

It would also be good if there was an explanation of the columns in the other csv files. For instance in the mpnn_design_stats csv a lot of columns have parameters numbered 1-5 (e.g. 1_pLDDT, 2_pLDDT, ..). What do these refer to? 

Finally, there are some seemingly random pdbs being saved in my MPNN/ and Trajectory/ folders (In this run I have three with the same base name, none of which is present in the Accepted outputs). Is there any significance to pdbs that end up saved in these locations rather than in the subfolders within these main output folders (e.g. Trajectory/Clashing; Trajectory/LowConfidence; MPNN/Relaxed)?

**Comment by LennartNickel** [comment]
Hey :) So the higher numbers will come from two reasons. First, there are early termination signals for trajectories if early on they don't look like they will result in something meaningful. Therefore, you run much more trajectories then what you see in the trajectory file. Second, it also checks for failures in the mpnn stage, which is by default 20x sequences/entries per trajectory. Therefore the higher numbers. 

The numbers before the csv files are referring to the af2 models used to calculate this metric. There is also an average of this metric. 

These MPNN files have not made it through the filtering before the relaxation (only sequences passing the af2 filters, are relaxed and checked for Rosetta metrics)

Hope that helps!

**Comment by samuelhaysom** [comment]
Thanks for this reply

So in the failure_csv file, if I have this right the following would be true: 

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SAMUEL~1.HAY/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SAMUEL~1.HAY/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


Trajectory_logits_pLDDT | Trajectory_softmax_pLDDT | Trajectory_one-hot_pLDDT | Trajectory_final_pLDDT | Trajectory_Contacts | Trajectory_Clashes | Trajectory_WrongHotspot
-- | -- | -- | -- | -- | -- | --
21 | 11 | 31 | 35 | 0 | 64 | 0



</body>

</html>


These indicate the number failing at end of each stage (so 21 failed at end of logits stage due to pLDDT below threshold, 64 failed at end of the trajectory stage due to clashes above the specified clashes threshold) 


<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SAMUEL~1.HAY/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SAMUEL~1.HAY/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


MPNN_score | MPNN_seq_recovery | pLDDT | pTM | i_pTM | pAE | i_pAE | i_pLDDT | ss_pLDDT
-- | -- | -- | -- | -- | -- | -- | -- | --
0 | 0 | 449 | 2291 | 3336 | 0 | 3379 | 0 | 0



</body>

</html>


These then indicate the number of MPNN sequences failing to pass the AF2 filters. So for instance after making 20 MPNN sequences per design that has passed the trajectory stage, then running 2 lots of AF2 runs on each one (?), 3336 of the resulting designs did not have an i_pAE score passing the threshold and 3379 of the resulting designs did not have a i_pAE score passing the threshold.

I've also just realised that the headers in the failure_csv are the same as the top level names in the default_filters.json file, so this is where the thresholds come from. Also this is why a lot of columns have 0 designs failing, because these metrics are set to null in the default_filters.json and so are not being considered during the design process.

**Comment by martinpacesa** [comment]
Indeed!
